src/screens/EventScreens/EventDetailScreen/index.js

```diff
@@ -23,21 +26,39 @@ import { useAuthStore } from '../../../shared/zustand/auth';
 import EventMemberItem from '../components/EventMemberItem';
 import Toast from 'react-native-toast-message';
 import { TOAST_UP_OFFSET } from '../../../components/constants';
+import { RectButton } from 'react-native-gesture-handler';
 
 
 const EventDetailScreen = ({ navigation, route }) => {
-  const { isLoading, fetchEventUserInfo ,currentEventUserInfo, cancelEvent, quitEvent, errorMessage } = useEventStore();
+  const { isLoading, fetchEventUserInfo ,currentEventUserInfo, cancelEvent, quitEvent, errorMessage, updateEvent } = useEventStore();
   const { userInfo } = useAuthStore();
   const [eventName, setEventName] = React.useState(route.params.eventName);
   const [eventLocation, setEventLocation] = React.useState(route.params.eventLocation);
   const [eventDescription, setEventDescription] = React.useState(route.params.eventDescription);
   const [eventStartTime, setEventStartTime] = React.useState(route.params.eventStartTime);
   const [eventEndTime, setEventEndTime] = React.useState(route.params.eventEndTime);
   const [isEditing, setIsEditing] = React.useState(false);
+  const [warning, setWarning] = React.useState(false);
+  const [enableSave, setEnableSave] = React.useState(true);
   React.useEffect(() => {
     fetchEventUserInfo(route.params.eventId);
   }, []);
-
+  React.useEffect(() => {
+    if (eventEndTime - eventStartTime <= 0) {
+      setWarning(true);
+    }
+    else {
+      setWarning(false);
+    }
+  })
+  React.useEffect(()=>{
+    if (!warning && (eventName!=='') && (eventLocation!=='')) {
+      setEnableSave(true);
+    }
+    else {
+      setEnableSave(false);
+    }
+  })
```

Comment:
add `eventName`, `eventLocation`, and `warning` to the dependency list, otherwise, this effect will only execute once.